### PR TITLE
Improve comment preserving coverage from 89% to 99%

### PR DESCRIPTION
**Pull request by MasseGuillaume**
Here is my workflow:

1) Run the PlaygroundSuite with `cover = true` to get the next failing slow test

https://github.com/scalacenter/scala-syntax/blob/c7cff63a19dce7214dcc2876f2627fcdb056193c/tests/unit/src/test/scala/scala/meta/internal/prettyprinters/PlaygroundSuite.scala

`unit/testOnly scala.meta.internal.prettyprinters.PlaygroundSuite`

2) open `output/input.scala` and `output/obtained.scala` and see where the comment is missing

3) copy `output/input.scala` into `tests/unit/src/test/resources/playground.scala` and minimize the example

4) set `cover = false` and run `unit/testOnly scala.meta.internal.prettyprinters.PlaygroundSuite`

5) add the missing t.`(` or t.`symbol`, etc in TreePrinter.scala and the extension method in SyntaxToken.scala

6) copy the minimize example from playground.scala to comments2.scala

### CODE REVIEW

File: tests/unit/src/test/resources/comments2.scala [hunk 1]
```diff
@@ -172,4 +172,9 @@ object A {
   {
 
   } 
+
+  a 
+  // L Term.Assign =
+  = // T Term.Assign =
```
Comment (olafurpg) [inline tests/unit/src/test/resources/comments2.scala:178]: the places comments can appear 😂

File: tests/unit/src/test/resources/comments2.scala [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  // Type.Apply L ]
+]
+
+object A {
+  { 
+    case _ => // Case T =>
+      ()
+  }
+
+  val a
```
Comment (olafurpg) [inline tests/unit/src/test/resources/comments2.scala:139]: No matching test for `var a`?

File: format/src/main/scala/scala/meta/internal/prettyprinters/TreePrinter.scala [hunk 3]
```diff
@@ -476,55 +503,88 @@ class TreePrinter private ()(implicit val trivia: AssociatedTrivias)
...
             Nil
         ) + print(t.templ)
       case t: Pkg.Object =>
-        val ddefn =
-          dDef(t.mods, `package` + space + `object`, print(t.name), Nil, Nil)
-        ddefn + print(t.templ)
+        spaceSeparated(
+          List(
+            dMods(t.mods),
+            `package` + space + `object`,
```
Comment (olafurpg) [inline format/src/main/scala/scala/meta/internal/prettyprinters/TreePrinter.scala:548]: what happens with `package /* BOOM */ object`? 😄 

People deserve to loose those comments 😈

**Review by olafurpg — APPROVED** [review]
Amazing work @MasseGuillaume ! Reaching >99% is a major milestone, formalizing `AssociatedTrivias` is a great contribution and the new comment tests are super valuable for the future ❤️ 

The commit log was easy to review, no major comments from me. This change LGTM 👍

File: tests/unit/src/test/resources/comments2.scala (hunk 1 above)
Comment (MasseGuillaume) [reply tests/unit/src/test/resources/comments2.scala:178]:
```scala
a = // T
  1
```

File: format/src/main/scala/scala/meta/internal/prettyprinters/TreePrinter.scala (hunk 3 above)
Comment (MasseGuillaume) [reply format/src/main/scala/scala/meta/internal/prettyprinters/TreePrinter.scala:548]: That's the remaining 1%.

**Pull request merged**